src/page-components/components/signUpForm/SignUpForm.jsx

```diff
@@ -1,58 +1,73 @@
-import { useEffect, useState } from "react";
-import { useRouter } from "next/navigation";
+import { useForm } from "react-hook-form";
+import { useRouter } from "next/router";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
 import Link from "next/link";
-import styles from "./SignUpForm.module.css";
 
-export default function SignUpForm() {
-  const [user, setUser] = useState({
-    firstName: "",
-    lastName: "",
-    email: "",
-    password: "",
-    confirmPassword: "",
-  });
-  const [buttonDisabled, setButtonDisable] = useState(true);
-  const router = useRouter();
+const validateEmail = (value) => {
+  const regex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  if (!regex.test(value)) {
+    return "Please enter a valid email address.";
+  }
+  return true;
+};
 
-  useEffect(() => {
-    if (
-      user.firstName.length > 0 &&
-      user.lastName.length > 0 &&
-      user.email.length > 0 &&
-      user.password.length > 0 &&
-      user.confirmPassword.length > 0
-    ) {
-      setButtonDisable(false);
-    } else {
-      setButtonDisable(true);
-    }
-  }, [user]);
+const validatePassword = (value) => {
+  if (value.length < 6) {
+    return "Password must be at least 6 characters.";
+  }
+  return true;
+};
 
-  const handleSubmit = async (e) => {
-    e.preventDefault();
+const validateConfirmPassword = (value, getValues) => {
+  if (value !== getValues().password) {
+    return "Passwords do not match.";
+  }
+  return true;
+};
 
-    const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
-    if (!emailPattern.test(user.email)) {
-      alert("Please enter a valid email address.");
-      return;
-    }
+export default function SignUpForm() {
+  const router = useRouter();
+  const form = useForm({
+    defaultValues: {
+      firstName: "",
+      lastName: "",
+      email: "",
+      password: "",
+      confirmPassword: "",
+    },
+    mode: "onSubmit",
+  });
 
-    if (user.confirmPassword != user.password) {
-      alert("Passwords don't match");
-      return;
-    }
+  const {
+    handleSubmit,
+    control,
+    formState: { errors },
+    getValues,
+  } = form;
 
+  async function onSubmit(values) {
     try {
       const res = await fetch("http://localhost:3000/api/users/signup", {
```

Comment:
Could use `/api/users/signup` instead of `http://localhost:3000/api/users/signup`